Source/WebKit/UIProcess/WebPageGroup.cpp

```diff
@@ -55,22 +55,26 @@ Ref<WebPageGroup> WebPageGroup::create(const String& identifier)
 
 static WebPageGroupData pageGroupData(const String& identifier)
 {
-    WebPageGroupData data;
-
     static NeverDestroyed<HashMap<String, PageGroupIdentifier>> map;
-    if (HashMap<String, PageGroupIdentifier>::isValidKey(identifier)) {
-        data.pageGroupID = map.get().ensure(identifier, [] {
-            return PageGroupIdentifier::generate();
-        }).iterator->value;
-    } else
-        data.pageGroupID = PageGroupIdentifier::generate();
+    auto pageGroupID = [&] {
+        if (HashMap<String, PageGroupIdentifier>::isValidKey(identifier)) {
+            return map.get().ensure(identifier, [] {
+                return PageGroupIdentifier::generate();
+            }).iterator->value;
+        }
+        return PageGroupIdentifier::generate();
+    }();
 
+    String validIdentifier;
     if (!identifier.isEmpty())
-        data.identifier = identifier;
+        validIdentifier = identifier;
     else
-        data.identifier = makeString("__uniquePageGroupID-"_s, data.pageGroupID.toUInt64());
+        validIdentifier = makeString("__uniquePageGroupID-"_s, pageGroupID.toUInt64());
```

Comment:
I saw in `ObjectIdentifier.h` that `toUInt64()` calls '`toRawValue()` and there's a comment that says "Use toRawValue instead". I wanted to ask if we prefer directly using `toRawValue()` in cases like this?

Reply:
I think it is wishful thinking at this point. Most call sites use toUint64() and ObjectIdentifier is an integer 99.9% of the time. If someone really wants to drop toUint64(), then they’ll just to a mass rename.

Reply:
Ok got it--that makes sense.